### ISSUE TITLE
Worth temporarily increasing the timeout limit when accessing through R?

### ISSUE THREAD

**Issue by jsayers-bsc** [MotusDev/Motus-TO-DO#719]
We've all noticed that the server has been pretty bogged down and slow lately, as with the internet in general. 

This has also created a lot of timeout issues when pulling in data through the R package. Anytime the timeout occurs in R the download has to be manually restarted. And it often seems to throw a lot of other errors that are not obviously related, but tend to follow a time-out (there have been quite a few questions from confused users as well).

Is there any value in temporary increasing the timeout limit from 120 sec to avoid this (while the general internet/server slowdown continues to be an issue)? Or does that just create as many issues as it may solve?

**Comment by steffilazerte** [comment]
In the R package, the timeout is already 120s... is there a server timeout too? Right now this timeout argument isn't available to the user, but I could make it so, if it's a package problem, so users could up the limit if they run into problems.

One of the timeout errors should be fixed in motus v.3.1.0 (it should automatically re-try at least once which might alleviate some of the problems).

**Comment by denislepage** [comment]
I suspect that the upgrades that I made to the delete procedure will make this much better for the time being.

Increasing the server timeout could potentially only exacerbate the problem, so it’s not always a panacea in any case. If the server is consistently not responding well, there’s probably an underlying issue that we need to resolve, like in this case.  The delete proc has been responsible for the few occasions where this has happened in the past few months, in all instances because it started looping endlessly without actually doing anything useful. The new approach I implemented should now hopefully totally avoid this.

If someone has a solution to the slow internet, please let me know!

**Comment by jsayers-bsc** [comment]
Yes I've been finding it times out quite a bit lately even at 120 sec. Normally that's sufficient but in the last few weeks I getting the time out quite a bit. 3 times in the last 5 minutes or so just now...

It does seem like it tries again, at least once, so that seems to be happening.

Looking back on the last several error messages, it actually looks like there are at least two different time out outcomes I've been noticing.

- What I assume is the expected timeout process, with the following message.
```
The server did not respond within 120s. Trying again...
Error: The server is not responding, please try again later.
```
- Another timeout message which gives this error and message:
```
The server did not respond within 120s. Trying again...
Error in if (stringr::str_detect(resp, "Timeout was reached")) { : 
  missing value where TRUE/FALSE needed
In addition: Warning message:
In stri_detect_regex(string, pattern, negate = negate, opts_regex = opts(pattern)) :
  argument is not an atomic vector; coercing
```

**Comment by steffilazerte** [comment]
Yeah that second one *should* be fixed on the beta3 branch (v3.1.0)

**Comment by steffilazerte** [comment]
That second error, actually occurs when the retry *works* but unfortunately the error prevents it from continuing. So when that is fixed some of the timeouts will result in a successful restart, but if you get two 2min timeouts in a row it'll stop (which I think makes sense?).
